controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/routing/RoutingApiHandler.java

```diff
@@ -0,0 +1,172 @@
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Level;
+
+/**
+ * This implements the /routing/v1 API, which provides operator with global routing control at both zone- and
+ * deployment-level.
+ *
+ * @author mpolden
+ */
+// TODO(mpolden): Add support for zones/deployments using rotations.
+public class RoutingApiHandler extends AuditLoggingRequestHandler {
+
+    private final Controller controller;
+
+    public RoutingApiHandler(Context ctx, Controller controller) {
+        super(ctx, controller.auditLogger());
+        this.controller = Objects.requireNonNull(controller, "controller must be non-null");
+    }
+
+    @Override
+    public HttpResponse auditAndHandle(HttpRequest request) {
+        try {
+            var path = new Path(request.getUri());
+            switch (request.getMethod()) {
+                case GET: return get(path);
+                case POST: return post(path);
+                case DELETE: return delete(path);
+                default: return ErrorResponse.methodNotAllowed("Method '" + request.getMethod() + "' is not supported");
+            }
+        } catch (IllegalArgumentException e) {
+            return ErrorResponse.badRequest(Exceptions.toMessageString(e));
+        } catch (RuntimeException e) {
+            log.log(Level.WARNING, "Unexpected error handling '" + request.getUri() + "'", e);
+            return ErrorResponse.internalServerError(Exceptions.toMessageString(e));
+        }
+    }
+
+    private HttpResponse delete(Path path) {
+        if (path.matches("/routing/v1/inactive/tenant/{tenant}/application/{application}/instance/{instance}/environment/{environment}/region/{region}")) return setDeploymentStatus(path, true);
+        if (path.matches("/routing/v1/inactive/environment/{environment}/region/{region}")) return setZoneStatus(path, true);
+        return ErrorResponse.notFoundError("Nothing at " + path);
+    }
+
+    private HttpResponse post(Path path) {
+        if (path.matches("/routing/v1/inactive/tenant/{tenant}/application/{application}/instance/{instance}/environment/{environment}/region/{region}")) return setDeploymentStatus(path, false);
+        if (path.matches("/routing/v1/inactive/environment/{environment}/region/{region}")) return setZoneStatus(path, false);
+        return ErrorResponse.notFoundError("Nothing at " + path);
+    }
+
+    private HttpResponse get(Path path) {
+        if (path.matches("/routing/v1/status/tenant/{tenant}/application/{application}/instance/{instance}/environment/{environment}/region/{region}")) return deploymentStatus(path);
+        if (path.matches("/routing/v1/status/environment/{environment}/region/{region}")) return zoneStatus(path);
+        return ErrorResponse.notFoundError("Nothing at " + path);
+    }
+
+    private HttpResponse setZoneStatus(Path path, boolean in) {
+        var zone = zoneFrom(path);
+        var status = in ? GlobalRouting.Status.in : GlobalRouting.Status.out;
```

Comment:
Nit: wouldn't it be more succinct if the parameter to this method was the enum constant? You could just print it(s name) in the output, as well.

Reply:
Maybe, will consider it in next PR. Realised I have to resolve the TODO I added here now anyway. 😞